File: rbuilder-tests/ecommerce-python/app/correctness.py
"""Intentional call graph for rBuilder expected-facts checks."""


def correctness_leaf() -> int:
    """Leaf — no outbound application calls."""
    return 42


def correctness_mid() -> int:
    """Mid — calls correctness_leaf."""
    return correctness_leaf() + 1


def correctness_root(flag: bool) -> int:
    """Root — calls correctness_mid and branches for a non-trivial CFG."""
    value = correctness_mid()
    if flag:
        return value * 2
    return value

File: rbuilder-tests/ecommerce-rust/src/correctness.rs
//! Intentional call graph for rBuilder expected-facts checks.
//! Prefer direct function calls so extraction does not depend on DI.

#![allow(dead_code)]

/// Leaf — no outbound application calls.
pub fn correctness_leaf() -> i32 {
    42
}

/// Mid — calls [`correctness_leaf`].
pub fn correctness_mid() -> i32 {
    correctness_leaf() + 1
}

/// Root — calls [`correctness_mid`] and branches for a non-trivial CFG.
pub fn correctness_root(flag: bool) -> i32 {
    let value = correctness_mid();
    if flag {
        value * 2
    } else {
        value
    }
}
